addon/mixins/sortable-item.js

```diff
@@ -328,6 +330,81 @@ export default Mixin.create({
     this._tellGroup('prepare');
     this.set('isDragging', true);
     this.sendAction('onDragStart', this.get('model'));
+
+    this._scrollOnEdges(drag);
+  },
+
+  /**
+    The maximum scroll speed when dragging element.
+    @property maxScrollSpeed
+    @default 20
+   */
+  maxScrollSpeed: 20,
+
+  _scrollOnEdges(drag) {
+    let groupDirection = this.get('group.direction');
+    let $element = this.$();
+    let scrollContainer = new ScrollContainer(scrollParent($element)[0]);
+    let itemContainer = {
+      width: $element.width(),
+      height: $element.height(),
+      get left() {
+        return $element.offset().left;
+      },
+      get right() {
+        return this.left + this.width;
+      },
+      get top() {
+        return $element.offset().top;
+      },
+      get bottom() {
+        return this.top + this.height;
+      }
+    };
+
+    let leadingEdgeKey, trailingEdgeKey, scrollKey;
+    if (groupDirection === 'x') {
+      leadingEdgeKey = 'left';
+      trailingEdgeKey = 'right';
+      scrollKey = 'scrollLeft';
+    } else {
+      leadingEdgeKey = 'top';
+      trailingEdgeKey = 'bottom';
+      scrollKey = 'scrollTop';
+    }
+
+    // Set a trigger padding that will start scrolling
+    // the box when the item reaches within padding pixels
+    // of the edge of the scroll container.
+    let checkScrollBounds = () => {
+      let leadingEdge = itemContainer[leadingEdgeKey];
+      let trailingEdge = itemContainer[trailingEdgeKey];
+      let scroll = scrollContainer[scrollKey]();
+
+      let delta = 0;
+      if (trailingEdge >= scrollContainer[trailingEdgeKey]) {
+        delta = trailingEdge - scrollContainer[trailingEdgeKey];
+      } else if (leadingEdge <= scrollContainer[leadingEdgeKey]) {
+        delta = leadingEdge - scrollContainer[leadingEdgeKey];
+      }
+
+      if (delta !== 0) {
```

Comment:
Sorry, reading the wrong thing! This prevents too many calls to expensive functions